cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -238,17 +241,13 @@ public void start() {
   /**
    * CDAP-6644 for secure impersonation to work,
    * we want other users to be able to write to the "path" directory,
-   * currently only cdap.user has read-write permissions while other users can only read the "/cdap/{path}" dir,
+   * currently only cdap.user has read-write permissions
+   * while other users can only read the "{hdfs.namespace}/{path}" dir,
    * we want to let others to be able to write to "path" directory, till we have a better solution.
    */
   private void createDirectory(String path) {
-    try {
-      String namespacedPath = String.format("/%s/%s", cConf.get(Constants.ROOT_NAMESPACE), path);
-      FileContext fileContext = FileContext.getFileContext(hConf);
-      createDirectory(fileContext, namespacedPath);
-    }  catch (UnsupportedFileSystemException e) {
-      LOG.error("Unsupported FileSystem Exception while trying to create directory", e);
-    }
+    String namespacedPath = String.format("%s/%s", cConf.get(Constants.CFG_HDFS_NAMESPACE), path);
```

Comment:
FileContext uses currentUser to run the operations, but we want to run as "hdfs.user". master is run as cdap user, so if hdfs.user is also cdap it would work. for other scenarios where the "hdfs.user" is different we would still try to create as "cdap" (currently running user) if we did `new FileContextProvider(cConf, hConf).get();` currently the FileContextProvider has wrapper code to run as "hdfs.user". that's why we decided to use that. if we don't want to use injector to get FileContext, we might have to repeat that logic.